lib/http2/scheduler.c

```diff
@@ -412,3 +412,21 @@ int h2o_http2_scheduler_is_active(h2o_http2_scheduler_node_t *root)
 {
     return root->_queue != NULL && !queue_is_empty(root->_queue);
 }
+
+h2o_http2_scheduler_node_t *h2o_http2_scheduler_find_new_parent(h2o_http2_scheduler_node_t *root, uint16_t new_weight)
```

Comment:
How about renaming the function to `h2o_http2_scheduler_find_parent_by_weight` or something?

As correctly pointed out in the comment, this logic is specific to how weight is used by a particular client implementation, and I think it'd be better if the name of the function implies that.

Reply:
Good point. Actually I have been struggling to come up with a better name, as this logic itself is extremely client (tree shape) dependent, but at the same time I did not want to make the name too lengthy. Let's go with `h2o_http2_scheduler_find_parent_by_weight` for now.

Reply:
Done.